src/localscope.h

```diff
@@ -13,12 +13,12 @@ class LocalScope
 	size_t numElements() const { return assignments.size() + children.size(); }
 	void print(std::ostream &stream, const std::string &indent, const bool inlined = false) const;
 	std::vector<class AbstractNode*> instantiateChildren(const std::shared_ptr<Context> &evalctx) const;
-	void addChild(shared_ptr<ASTNode> astnode);
+	void addChild(shared_ptr<ASTNode> &&astnode);
 private:
-	void addModuleInst(shared_ptr<class ModuleInstantiation> astnode);
-	void addModule(shared_ptr<class UserModule> module);
-	void addFunction(shared_ptr<class UserFunction> function);
-	void addAssignment(shared_ptr<class Assignment> assignment);
+	void addModuleInst(shared_ptr<class ModuleInstantiation> &&astnode);
+	void addModule(shared_ptr<class UserModule> &&module);
+	void addFunction(shared_ptr<class UserFunction> &&function);
+	void addAssignment(shared_ptr<class Assignment> &&assignment);
```

Comment:
With your latest changes, I think these should go back to pass-by-value.

Reply:
The latest changes were only on calls from `parser.y` to `LocalScope::addChild`,  I made these all private so they should only be called directly by addChild, which still does `std::move` on the specific `dynamic_pointer_cast` variables.

So, I think it would make sense to change addChild back to by-value now, but not these?

Reply:
Or could the calls in addChild also be changed in some way to avoid std::move?

Reply:
ah, I overlooked that. You're right - these are private methods and it all makes sense now :)

..so just changing addChild() to by-value should be good!